android/app/src/main/java/app/organicmaps/car/screens/NavigationScreen.java

```diff
@@ -31,28 +39,34 @@
 import app.organicmaps.car.screens.base.BaseMapScreen;
 import app.organicmaps.location.LocationHelper;
 import app.organicmaps.location.LocationListener;
+import app.organicmaps.routing.NavigationService;
 import app.organicmaps.routing.RoutingController;
 import app.organicmaps.routing.RoutingInfo;
 import app.organicmaps.sound.TtsPlayer;
+import app.organicmaps.util.LocationUtils;
 import app.organicmaps.util.log.Logger;
 
 import java.util.List;
 import java.util.Objects;
 
-public class NavigationScreen extends BaseMapScreen implements RoutingController.Container, NavigationManagerCallback, LocationListener
+public class NavigationScreen extends BaseMapScreen implements RoutingController.Container, NavigationManagerCallback
 {
   private static final String TAG = NavigationScreen.class.getSimpleName();
 
+  public static final String MARKER = NavigationScreen.class.getSimpleName();
+
   @NonNull
   private final RoutingController mRoutingController;
   @NonNull
   private final NavigationManager mNavigationManager;
+  @NonNull
+  private final LocationListener mLocationListener = (unused) -> updateTrip();
 
-  public NavigationScreen(@NonNull CarContext carContext, @NonNull SurfaceRenderer surfaceRenderer)
+  private NavigationScreen(@NonNull Builder builder)
```

Comment:
No, it's not.
The builder is essential because it has additional logic for marking NavigationScreen with `scree.setMarker(MARKER)`.

You'll have to write 3 lines of code in each place the NavigationScreen is created instead of 1 now